src/main/scala/ch/epfl/bluebrain/nexus/kg/storage/S3StorageOperations.scala

```diff
@@ -159,11 +158,25 @@ object S3StorageOperations {
       }
   }
 
-  private def getKey(bucket: String, path: Path): Option[String] = path match {
-    case Slash(Segment(head, Slash(tail))) if head == bucket =>
-      Some(URLDecoder.decode(tail.toString, "UTF-8"))
-    case _ =>
-      logger.error(s"Error decoding key from S3 object URI '$path' in bucket '$bucket'")
-      None
+  private def getKey(storage: S3Storage, location: Uri): Option[String] = {
+    val path = storage.settings.endpoint match {
```

Comment:
you could use collect to make it shorter. Something like that should work:
```scala
val path = storage.settings.endpoint.collect {
   case Some(endpoint) if location.toString.startsWith(endpoint) || endpoint.startsWith("http://localhost") => location.path
   case None if location.authority.toString.endsWith("amazonaws.com") => location.path
}
```

Reply:
`collect` won't work if the optional endpoint is empty